qiskit_nature/algorithms/ground_state_solvers/adapt_vqe.py

```diff
@@ -58,28 +72,44 @@ def __init__(
             delta: the finite difference step size for the gradient computation. It has a minimum
                 value of 1e-5.
             max_iterations: the maximum number of iterations of the AdaptVQE algorithm.
+            gradient: a class that converts operator expression to the first-order gradient based
+                on the method mentioned.
         """
         validate_min("threshold", threshold, 1e-15)
         validate_min("delta", delta, 1e-5)
 
+        if gradient is None:
+            gradient = Gradient(grad_method="param_shift")
+
         super().__init__(qubit_converter, solver)
 
         self._threshold = threshold
         self._delta = delta
         self._max_iterations = max_iterations
+        self._gradient = gradient
 
         self._excitation_pool: List[OperatorBase] = []
         self._excitation_list: List[OperatorBase] = []
 
         self._main_operator: PauliSumOp = None
         self._ansatz: QuantumCircuit = None
 
+    @property
+    def gradient(self) -> Optional[GradientBase]:
+        """Returns the gradient."""
+        return self._gradient
+
+    @gradient.setter
+    def gradient(self, grad: Optional[GradientBase]):
+        """Sets the gradient."""
+        self._gradient = grad
+
     def returns_groundstate(self) -> bool:
         return True
 
     def _compute_gradients(
         self,
-        theta: List[float],
+        theta: List[Float],
```

Comment:
```suggestion
        theta: List[float],
```
As per comment above